### PR TITLE
Add status check for README.md files

### PR DESCRIPTION
**Pull request by duffh**
## Description

Added github actions to check readme formatting and readme consistency across platforms. 

## Type of change

- [x] Other enhancement

## Checklist

- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] All variable and method names are good and make sense

### CODE REVIEW

File: tools/CI/README_Metadata_SampleSyncCheck/sample_sync.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import sys
```
Comment (ZackAllen) [inline tools/CI/README_Metadata_SampleSyncCheck/sample_sync.py:1]: I think we should rename this script to differentiate it from the other script.

File: tools/CI/README_Metadata_SampleSyncCheck/entry.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import subprocess as sp
+
+def main():
+    msg = 'Entry point of the docker to run the sample-sync script.'
+    parser = argparse.ArgumentParser(description=msg)
+    parser.add_argument('-s', '--string', help='A JSON array of file paths.')
+    args = parser.parse_args()
+    
+    print("** Starting sample sync **")
+    code2 = sp.call(f'python3 /sample_sync.py',  shell=True)
```
Comment (ZackAllen) [inline tools/CI/README_Metadata_SampleSyncCheck/entry.py:14]: is code2 used?

File: tools/CI/README_Metadata_SampleSyncCheck/sample_sync.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    # args = sys.argv[1]
+    
+    readme_script_path = os.path.join(".","tools", "readme_copy", "readme_copy.py")
+    metadata_script_path = os.path.join(".", "tools", "metadata_tools", "process_metadata.py")
+
+    print("Copying readmes")
+    subprocess.run(["python", readme_script_path])
+
+    print("Updating metadata")
+    subprocess.run(["python", metadata_script_path, "src"])
```
Comment (ZackAllen) [inline tools/CI/README_Metadata_SampleSyncCheck/sample_sync.py:18]:
```suggestion
    subprocess.run(["python", metadata_script_path])
```
IIRC the argument wasn't actually necessary here, something else was the problem at the time.

File: tools/CI/README_Metadata_StyleCheck/README_style_checker.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
+#!/usr/bin/env python3
+
+"""
+Comments in PyCharm style.
+
+References
+
+- Tag sorter by Zack
+  /common-samples/blob/master/tools/net/tag_sorter/tag_sorter.py
```
Comment (ZackAllen) [inline tools/CI/README_Metadata_StyleCheck/README_style_checker.py:9]: References to common-samples shouldnt go in our public repo.

File: tools/metadata_tools/file_utils.py [hunk 5]
```diff
@@ -12,8 +12,12 @@ def safe_read_contents(path_to_file):
             with open(path_to_file, "r", encoding='utf-8') as handle:
                 original_contents = handle.read()
         except UnicodeDecodeError:
-            with open(path_to_file, "r", encoding='utf-16') as handle:
-                original_contents = handle.read()
+            try:
+                with open(path_to_file, "r", encoding='utf-16') as handle:
+                    original_contents = handle.read()
+            except:
+                print(path_to_file)
```
Comment (ZackAllen) [inline tools/metadata_tools/file_utils.py:19]: Should probably add an informative message to this.
```suggestion
                print("Error reading file: "+path_to_file)
```

File: tools/metadata_tools/file_utils.py [hunk 6]
```diff
@@ -28,5 +32,8 @@ def safe_write_contents(path_to_file, new_content):
...
                 rewrite_handle.write(new_content)
         except UnicodeEncodeError:
-            with open(path_to_file, 'w+', encoding="utf-16") as rewrite_handle:
-                rewrite_handle.write(new_content)
\ No newline at end of file
+            try:
+                with open(path_to_file, 'w+', encoding="utf-16") as rewrite_handle:
+                    rewrite_handle.write(new_content)
+            except:
+                print(path_to_file)
```
Comment (ZackAllen) [inline tools/metadata_tools/file_utils.py:39]:
```suggestion
                print("Error writing file: "+path_to_file)
```

File: tools/metadata_tools/process_metadata.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
+from http.client import NETWORK_AUTHENTICATION_REQUIRED
```
Comment (ZackAllen) [inline tools/metadata_tools/process_metadata.py:1]: Line can be removed.

File: tools/metadata_tools/process_metadata.py [hunk 8]
```diff
@@ -82,25 +82,34 @@ def write_samples_toc(platform_dir, relative_path_to_samples, samples_in_categor
...
+
+        samples = list(samples_in_categories[category])
+        samples.sort(key=lambda s: s.friendly_name)
+        for sample in samples:
             entry_url = f"{relative_path_to_samples}/{formal_category}/{sample.formal_name}"
             entry_url = urllib.parse.quote(entry_url)
             readme_text += f"* [{sample.friendly_name}]({entry_url}) - {sample.description}\n"
         readme_text += "\n"
-    
+    readme_text = readme_text[:-1]
```
Comment (ZackAllen) [inline tools/metadata_tools/process_metadata.py:100]:
```suggestion
    # Remove trailing newline character from readme.
    readme_text = readme_text[:-1]   
```

File: tools/metadata_tools/process_metadata.py [hunk 9]
```diff
@@ -185,7 +195,9 @@ def update_attribute(sample, sample_dir):
             file.seek(0)
             file.write(''.join(lines))
             file.close()
-    except:
+
+    except Exception as e:
+        print(e)
```
Comment (ZackAllen) [inline tools/metadata_tools/process_metadata.py:200]: Maybe combine print statements here.

File: tools/metadata_tools/sample_metadata.py [hunk 10]
```diff
@@ -3,7 +3,7 @@
 from distutils.dir_util import copy_tree
 from shutil import copyfile, rmtree
 import re
-import requests
+# import requests
```
Comment (ZackAllen) [inline tools/metadata_tools/sample_metadata.py:6]: We should delete every commented section in this.

**Review by ZackAllen — CHANGES_REQUESTED** [review]

**Comment by ZackAllen** [comment]
Saw this warning from the readme check, any idea how to resolve?
```
[Check README and metadata format for changed files](https://github.com/Esri/arcgis-runtime-samples-dotnet/runs/7135935518?check_suite_focus=true#step:2:7)
Received event from pull_request, but also received a before(87007f750e4e320573a0ee699a55345ca6ec533b) or after(5f6811ad5ff37f13955a974ae6262629c5551f5c) value.
 I am assuming you want to use a Push event but forgot something, so I'm giving you a message.
```

**Comment by duffh** [comment]
> Saw this warning from the readme check, any idea how to resolve?
> 
> ```
> [Check README and metadata format for changed files](https://github.com/Esri/arcgis-runtime-samples-dotnet/runs/7135935518?check_suite_focus=true#step:2:7)
> Received event from pull_request, but also received a before(87007f750e4e320573a0ee699a55345ca6ec533b) or after(5f6811ad5ff37f13955a974ae6262629c5551f5c) value.
>  I am assuming you want to use a Push event but forgot something, so I'm giving you a message.
> ```

I can see that this warning is referenced in an open issue logged against the GitHub action we are using here. I don't think it's a deal breaker but it is a bit annoying. Looks like the developer has not fixed it in the past 2 years. We could use a different action or accept that this warning message will appear. What do you think?

https://github.com/trilom/file-changes-action/issues/97

**Comment by ZackAllen** [comment]
> > Saw this warning from the readme check, any idea how to resolve?
> > ```
> > [Check README and metadata format for changed files](https://github.com/Esri/arcgis-runtime-samples-dotnet/runs/7135935518?check_suite_focus=true#step:2:7)
> > Received event from pull_request, but also received a before(87007f750e4e320573a0ee699a55345ca6ec533b) or after(5f6811ad5ff37f13955a974ae6262629c5551f5c) value.
> >  I am assuming you want to use a Push event but forgot something, so I'm giving you a message.
> > ```
> 
> I can see that this warning is referenced in an open issue logged against the GitHub action we are using here. I don't think it's a deal breaker but it is a bit annoying. Looks like the developer has not fixed it in the past 2 years. We could use a different action or accept that this warning message will appear. What do you think?
> 
> [trilom/file-changes-action#97](https://github.com/trilom/file-changes-action/issues/97)

Not a big issue.

File: tools/CI/README_Metadata_StyleCheck/README_style_checker.py (hunk 4 above)
Comment (ZackAllen) [reply tools/CI/README_Metadata_StyleCheck/README_style_checker.py:9]: This is not resolved.

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**